### PR TITLE
[s] Fixes tritium rad_pulse overload

### PR DESCRIPTION
**Pull request by Buggy123**
🆑 
fix: Tritium no longer produces so much radiation that it crashes the server
/🆑 

tags: FUCK, priority:CRITICAL, atmospherics, fix, test merge candidate

This also slightly increases the rads produced because it also sets a minimum reaction energy and so that'll sometimes result in rads not being produced most of the time. The radiation produced is completely arbitrary and hasn't been balanced tested at all anyway.

closes #35660

### CODE REVIEW

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -13,7 +13,8 @@
 #define NITRYL_FORMATION_ENERGY				100000
 #define TRITIUM_BURN_OXY_FACTOR				100
 #define TRITIUM_BURN_TRIT_FACTOR			10
-#define TRITIUM_BURN_RADIOACTIVITY_FACTOR	1000000 //The neutrons gotta go somewhere. Completely arbitrary number.
+#define TRITIUM_BURN_RADIOACTIVITY_FACTOR	50000 //The neutrons gotta go somewhere. Completely arbitrary number.
```
Comment (lzimann) [inline code/modules/atmospherics/gasmixtures/reactions.dm]: this is going from 1 million to 50k, it's quite huge of a change

Comment (Buggy123) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: the rad proc is now rad(STUFF && prob(10)) so every burst needs to produce 10 times as much energy to compensate, 1m/50k=20 so in effect I just doubled it. That still might be a lot, but I'm serious when I said I didn't balance the radiation it makes a lot; I was just changing it up/down by factors of 10.

Actually, i probably should've left it one order of magnitude higher. When I was testing it I put trit+ox cans in the atmos mix chamber and ignited them while standing in them, to see how many rads I got. Tritum is *supposed* to produce enough radiation while burning to produce useful amounts of power, so in a ideal situation like that you'd probably expect to get about as many rads as sitting next to the SM for a good while. I'll do some more testing later.

**Comment by DaedalusGame** [comment]
If you actually want to test this, start up the profiler and then unleash an actual plasma fire. Burning chamber is 3 tiles, you want to check the effect when 500 tiles fire this reaction

**Comment by Cyberboss** [comment]
The speed isn't the problem, it's the volume of the radiation datums

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (arsserpentarium) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: I think, you should get more than gives SM.Because tritium is expends,SM not

**Comment by arsserpentarium** [comment]
also for performance reasons possibly you should make some value k, which  set probablity of pulses(prob(k*100)) and their power (TRITIUM_BURN_RADIOACTIVITY_FACTOR/k).So it will be more easy to balance performance wise and maybe variate that value as function of server load.

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (Buggy123) [reply code/modules/atmospherics/gasmixtures/reactions.dm]: Exactly. By calibrating it with a room of high-o2 tritium burnmix I probably set the radiation too low.

**Comment by Buggy123** [comment]
>also for performance reasons possibly you should make some value k, which set probablity of pulses(prob(k*100)) and their power (TRITIUM_BURN_RADIOACTIVITY_FACTOR/k).

sounds good

>So it will be more easy to balance performance wise and maybe variate that value as function of server load.

uuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh...

**Comment by DaedalusGame** [comment]
not really that helpful if rad pulses aren't pooled and you thus can't obtain information of how many there are.

**Comment by arsserpentarium** [comment]
no need to uhh.I said maybe.

**Comment by Supermichael777** [comment]
>uuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh...

My reaction when asked to fuck with tgui. HTML was a shitty method but at least a pleb like me had a shot of knowing that.

**Comment by Buggy123** [comment]
Merge conflict resolved.

**Comment by Buggy123** [comment]
Why did Travis fail? The only warning I can see in there is about the dirty varedit on Snowden.

**Comment by kevinz000** [comment]
testmerge this and let me test live if it doesn't crash it never will tbh

**Comment by kevinz000** [comment]
;^)

**Pull request merged**